MvvmCross/ViewModels/MvxNotifyTask.cs

```diff
@@ -31,7 +31,6 @@ private MvxNotifyTask(Task task, Action<Exception> onException)
         {
             try
             {
-                await Task.Yield();
```

Comment:
Why did you remove this `Task.Yield`? It was added in purpose.

Reply:
Two things:
- my understanding is that calling Yield will cause an unnecessary latency in the task being invoked
- if it is there for a reason (eg best practice or some other guidance) we should add it back with an appropriate comment and/or reference

Reply:
Is the point of the Yield call to force the code to run asynchronously? If so, this should be an option that's exposed when creating the MvxNotifyTask, rather than being the default.

Reply:
It was added here #2642 (also I don't see any mention to this change on this PR other than the change in the source code).

You're right about why is it there. The problem is that if the code isn't run asynchronously, there is a (quite high) chance the exception will be thrown before MvxNotifyTask starts monitoring the task. This is highly unexpected when you are delegating all error handling responsibilities to the object.

I agree this adds a (very) small latency to all operations running inside of it, but MvxNotifyTask is not just common async/await.